### DISCUSSION TITLE
Building docker-aware tasks

### DISCUSSION THREAD

**Discussion by gnilrets** [pyinvoke/invoke#870] [Q&A]
I would like to write an invoke task that would adjust it's behavior based on whether it's running in a docker container or not.

For example, suppose I had a simple task that ran black formatting:
```
@task
def format(ctx):
    ctx.run('black myfile.py')
```

When `inv format` is run within my docker container, it should simply run `black myfile.py`.  However, if I run `inv format` on the host machine, I'd like invoke to wrap that command in a docker run statement.  Suppose I have some environment variable that I can check to see if I'm running in a container (`IS_DOCKER`).  I could write something like this to accomplish what I want:

```
@task
def format(ctx):
    if os.environ.get('IS_DOCKER') == '1':
        ctx.run('black myfile.py')
    else:
        ctx.run('docker run --rm my-container "black myfile.py"')
```

I could probably make this a bit nicer by writing a run `run_cmd_maybe_docker` method.  However, it seems like there should be a more natural invoke-style way to accomplish this using contexts and runners.

**Comment by ab623** [comment]
I have the same issue. I just have a global variable which detects if i'm in a docker env or now.

I just pass the command like `black myfile.py'` into a helper function which reads the global variable and constructs and returns the correct command for me, which I pass to the run function.
